nilrt_snac/_configs/_auditd_config.py

```diff
@@ -0,0 +1,226 @@
+    subprocess.run(args, check=True)
+
+def ensure_proper_groups(groups: List[str]) -> None:
+    "Ensures the specified groups exist on the system."
+    for group in groups:
+        try:
+            grp.getgrnam(group)
+        except KeyError:
+            _cmd("groupadd", group)
+            logger.info(f"Group {group} created.")
+
+def format_email_template_text(audit_email: str) -> str:
+    return f"""
+    #!/usr/bin/perl
+    use strict;
+    use warnings;
+    use Net::SMTP;
+
+    # Configuration
+    my $smtp_server = 'smtp.yourisp.com';
+    my $smtp_user = 'your_email@domain.com';
+    my $smtp_pass = 'your_password';
+    my $from = 'your_email@domain.com';
+    my $to = '{audit_email}';
+    my $subject = 'Audit Alert';
+    my $body = "A critical audit event has been triggered: $ARGV[0]";
+
+    # Create SMTP object
+    my $smtp = Net::SMTP->new($smtp_server, Timeout => 60)
+        or die "Could not connect to SMTP server: $!";
+
+    # Authenticate
+    $smtp->auth($smtp_user, $smtp_pass)
+        or die "SMTP authentication failed: $!";
+
+    # Send email
+    $smtp->mail($from)
+        or die "Error setting sender: $!";
+    $smtp->to($to)
+        or die "Error setting recipient: $!";
+    $smtp->data()
+        or die "Error starting data: $!";
+    $smtp->datasend("To: $to\\n");
+    $smtp->datasend("From: $from\\n");
+    $smtp->datasend("Subject: $subject\\n");
+    $smtp->datasend("\\n");
+    $smtp->datasend("$body\\n");
+    $smtp->dataend()
+        or die "Error ending data: $!";
+    $smtp->quit;
+    """
+
+def is_valid_email(email: str) -> bool:
+    "Validates an email address."
+    email_regex = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+$'
+    return re.match(email_regex, email) is not None
+
+
+
+class _AuditdConfig(_BaseConfig):
+    def __init__(self):
+        self._opkg_helper = opkg_helper
+        self.log_path = os.path.realpath('/var/log')
+        self.audit_config_path = '/etc/audit/auditd.conf'
+
+    def configure(self, args: argparse.Namespace) -> None:
+        print("Configuring auditd...")
+        auditd_config_file = EqualsDelimitedConfigFile(self.audit_config_path)
+        dry_run: bool = args.dry_run
+
+        # Check if auditd is already installed
+        if not self._opkg_helper.is_installed("auditd"):
+            self._opkg_helper.install("auditd")
+        
+        #Ensure proper groups exist
+        groups_required = ["adm", "sudo"]
+        ensure_proper_groups(groups_required)
+
+        # Prompt for email if not provided
+        audit_email = args.audit_email
+        unattended_bypass = args.yes
+        if not audit_email:
+            audit_email = auditd_config_file.get("action_mail_acct")
+        if not is_valid_email(audit_email) and not unattended_bypass:
+            audit_email = input("Please enter your audit email address: ")
+        else:
+            # Use local default e-mail
+            audit_email = f"root@{socket.gethostname()}"
```

Comment:
Does this not work if you just set the `action_mail_acct = root`? The hostname suffix I think will force the email out onto the network, which is probably why you need that perl script. But we should just be able to route the email internally into the root account - after maybe installing an MTA like `postfix`.